lib/ansible/modules/windows/win_domain_membership.ps1

```diff
@@ -45,6 +45,23 @@ Function Get-DomainMembershipMatch {
 
         return $domain_match
     }
+    catch [System.Security.Authentication.AuthenticationException] {
+        Write-DebugLog "Checking to see if ansible_user account is local and attempt a different method."
+        Add-Type -AssemblyName System.DirectoryServices.AccountManagement            
+        $UserPrincipal = [System.DirectoryServices.AccountManagement.UserPrincipal]::Current
+        If ($UserPrincipal.ContextType -eq "Machine") {
+            $current_dns_domain = (Get-WmiObject -Class Win32_ComputerSystem).Domain
```

Comment:
The `*-Wmi` cmdlets have been deprecated in favour of the `*-Cim` cmdlets, you should also restrict the output to retrieve only the properties you need with `-Property`, e.g.

`(Get-CimInstance -ClassName Win32_ComputerSystem -Property Domain).Domain`